src/components/Calculator.js

```diff
@@ -1,46 +1,87 @@
 import React from 'react';
+import calculate from '../logic/calculate';
 
 class Calculator extends React.PureComponent {
+  constructor(props) {
+    super(props);
+    this.state = { total: 0, next: null, operation: null };
+  }
+
+  clickReceived = (event) => {
+    const buttonName = event.target.innerText;
+    const answer = calculate(this.state, buttonName);
+    this.setState(answer);
+  };
+
```

Comment:
- [ ] It is required to make your Calculator component functional component and make use of `useState && props` you've learned. Kindly implement those requirements in your project. :point_down: 

![project-rqts](https://user-images.githubusercontent.com/26597432/219049415-5811e314-1f1a-400d-825f-06c8a2693e23.png)